test/functional/p2p_filter.py

```diff
@@ -104,6 +105,20 @@ def run_test(self):
             txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 7)
             filter_node.wait_for_tx(txid)
 
+        self.log.info('Check that request for filtered blocks is ignored if no filter is set')
+        filter_node.merkleblock_received = False
+        filter_node.tx_received = False
+        with self.nodes[0].assert_debug_log(expected_msgs=['sending inv', 'received getdata']):
```

Comment:
I know I suggested to use `assert_debug_log` for the inv and getdata, but a better solution would be to use `wait_for_inv`, which also checks the hash:

```diff
diff --git a/test/functional/p2p_filter.py b/test/functional/p2p_filter.py
index 0b338caa9a..6a8f54e376 100755
--- a/test/functional/p2p_filter.py
+++ b/test/functional/p2p_filter.py
@@ -7,12 +7,13 @@ Test BIP 37
 """
 
 from test_framework.messages import (
+    CInv,
     MSG_BLOCK,
     MSG_FILTERED_BLOCK,
-    msg_getdata,
-    msg_filterload,
     msg_filteradd,
     msg_filterclear,
+    msg_filterload,
+    msg_getdata,
 )
 from test_framework.mininode import P2PInterface
 from test_framework.test_framework import BitcoinTestFramework
@@ -103,10 +104,11 @@ class FilterTest(BitcoinTestFramework):
         filter_node.merkleblock_received = False
         filter_node.tx_received = False
         with self.nodes[0].assert_debug_log(expected_msgs=['sending inv', 'received getdata'], timeout=10):
-            self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())
-        filter_node.sync_with_ping()
-        assert not filter_node.merkleblock_received
-        assert not filter_node.tx_received
+            h = self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())[0]
+            filter_node.wait_for_inv([CInv(MSG_FILTERED_BLOCK, int(h, 16))])
+            filter_node.sync_with_ping()
+            assert not filter_node.merkleblock_received
+            assert not filter_node.tx_received
 
         self.log.info('Check that sending "filteradd" if no filter is set is treated as misbehavior (+100)')
         assert_equal(self.nodes[0].getpeerinfo()[0]['banscore'], 0)

Reply:
Alright :+1: The `assert_debug_log` can then be removed completely? I force-pushed an update.

Reply:
I'd say to keep it for the 'received getdata', but no strong opinion

Reply:
Done.